src/utils.jl

```diff
@@ -88,54 +86,78 @@ function get_problem_data(model::MOI.AbstractOptimizer)
         func = MOI.get(model, MOI.ConstraintFunction(), con)
         set = MOI.get(model, MOI.ConstraintSet(), con)
 
-        G[i, :] = coefficient.(func.terms)'
+        for (j, var_idx) in enumerate(var_list)
+            for term in func.terms
+                if term.variable_index == var_idx
+                    G[i,j] = MOI.coefficient(term)
+                end
+            end
+        end
         h[i] = set.upper - func.constant
     end
-    
+
     # handle equality constraints
     eq_con_idx   = MOI.get(
-                        model, 
+                        model,
                         MOI.ListOfConstraintIndices{
-                            MOI.ScalarAffineFunction{Float64}, 
+                            MOI.ScalarAffineFunction{Float64},
                             MOI.EqualTo{Float64}
                         }())
     neq   = size(eq_con_idx)[1]
-    
+
     A = zeros(neq, nz)
     b = zeros(neq)
-    
+
     for i in 1:neq
         con = eq_con_idx[i]
 
         func = MOI.get(model, MOI.ConstraintFunction(), con)
         set = MOI.get(model, MOI.ConstraintSet(), con)
 
-        A[i, :] = coefficient.(func.terms)'
+        for x in func.terms
+            A[i, x.variable_index.value] = x.coefficient
+        end
         b[i] = set.value - func.constant
     end
 
-    
+
     # handle objective
     # works both for affine and quadratic objective functions
     objective_function = MOI.get(model, MOI.ObjectiveFunction{MOI.ScalarQuadraticFunction{Float64}}())
     Q = zeros(nz, nz)
-    
+    q = zeros(nz)
+
     if typeof(objective_function) == MathOptInterface.ScalarAffineFunction{Float64}
-        q = coefficient.(objective_function.terms)
+        for x in objective_function.terms
+            q[x.variable_index.value] = x.coefficient
+        end
     elseif typeof(objective_function) == MathOptInterface.ScalarQuadraticFunction{Float64}
-        # @assert size(objective_function.quadratic_terms)[1] == (nz * (nz + 1)) / 2    
-        
-        var_to_id = Dict(var_idx .=> 1:nz)
-        
+        # @assert size(objective_function.quadratic_terms)[1] == (nz * (nz + 1)) / 2
+
+        var_to_id = Dict(var_list .=> 1:nz)
+
         for quad in objective_function.quadratic_terms
             i = var_to_id[quad.variable_index_1]
             j = var_to_id[quad.variable_index_2]
             Q[i,j] = quad.coefficient
             Q[j,i] = quad.coefficient
         end
-        
-        q = coefficient.(objective_function.affine_terms)
+
+        q = MOI.coefficient.(objective_function.affine_terms)
+    end
+
+    return Q, q, G, h, A, b, nz, var_list, nineq, ineq_con_idx, neq, eq_con_idx
+end
+
+# might slow down computation
+# need to find a faster way
+function CSRToCSC(B::MatOI.SparseMatrixCSRtoCSC{Int64})
```

Comment:
In fact, it's easier to get the conic form for CSC A, we do it in CSR in MatOI as this is what SCS needs but we can add a mutable CSC matrix in MatOI too quite easily so that DiffOpt can use that but let's leave that for another PR

Reply:
maybe document it in an issue?